hydra-main/src/test/java/com/addthis/hydra/job/web/JobRequestHandlerImplTest.java

```diff
@@ -117,33 +117,22 @@ public void updateMinionType_notOnTargetType() throws Exception {
 
         try {
             impl.updateMinionType("job_id","newMinion", username, token, sudo);
+            fail("IllegalArgumentException expected but not thrown");
         } catch (IllegalArgumentException e) {
             // Expected
         }
-        verify(spawn, times(0)).updateJob(job);
-        assertSame(job.getMinionType(), "oldMinion");
+        verify(spawn, never()).updateJob(job);
+        assertEquals("oldMinion", job.getMinionType());
     }
 
-    @Test
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
This is the convention for verifying an expected exception. I also simplified the test. I think it's a better balance between test coverage and simplicity.